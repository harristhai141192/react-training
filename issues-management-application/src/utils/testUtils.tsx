// Libraries
import React from 'react';
import { ThemeProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

// Themes
import themes from '@themes/index';

const customRender = (component: React.ReactNode) => {
  return render(
    <ThemeProvider theme={themes}>
      <Router>{component}</Router>
    </ThemeProvider>,
  );
};

export * from '@testing-library/react';
export { customRender as render };
