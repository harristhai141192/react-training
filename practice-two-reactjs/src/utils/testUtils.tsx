import { ThemeProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import themes from '@themes/index';
import { ReactNode } from 'react';

const customRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={themes}>{component}</ThemeProvider>);
};

export * from '@testing-library/react';
export { customRender as render };
