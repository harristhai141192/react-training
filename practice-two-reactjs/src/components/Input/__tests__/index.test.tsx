// libraries
import { render, getByPlaceholderText, getByText, getByTestId } from '@utils/testUtils';
// components
import InputComponent from '../index';
import '@testing-library/jest-dom';

const baseProps = {
  error: false,
  placeholder: '',
};

const setup = (overrideProps = {}) => {
  const props = {
    ...baseProps,
    ...overrideProps,
  };
  return render(<InputComponent {...props} />);
};

describe('Component [Input] should match snapshot', () => {
  test('It should match snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  test('Component [Input] should render correctly', () => {
    const { container } = setup({ placeholder: 'Input Your Value' });
    const getPlaceHolder = getByPlaceholderText(container, 'Input Your Value');
    expect(getPlaceHolder).toBeInTheDocument();
  });

  test('Component [Input] should show error message when invalid', () => {
    const { container } = setup({
      placeholder: 'Input Your Value',
      error: true,
      errorText: 'Error!',
    });

    const inputError = getByText(container, 'Error!');
    expect(inputError).toBeInTheDocument();
  });

  test('Component [Input] should be red border when get error', () => {
    const { container } = setup({
      placeholder: 'Input Your Value',
      error: true,
      errorText: 'Error!',
    });

    expect(getByTestId(container, 'Input Test')).toHaveStyle('border-color: crimson');
  });
});
