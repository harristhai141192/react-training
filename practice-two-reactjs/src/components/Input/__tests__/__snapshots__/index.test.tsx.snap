// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component [Input] should match snapshot It should match snapshot 1`] = `
<div>
  <input
    class="chakra-input css-10eaxd9"
    placeholder=""
    value=""
  />
</div>
`;
